README.ideal.md

```diff
@@ -0,0 +1,131 @@
+_Ni-Mh — batteries included environments with Nickel inside_.
+
+Managing a project's development environment involves configuring a lot of tools and occasionnally getting to communicate with each other: One (or several) package manager, a CI system, a service manager (to run that local postgresql database without which you can't test anything), custom utility scripts that end up spawning everywhere, etc.
+Ni-Mh (called “nickel-nix” or “Nixel” at the moment, but I really like Ni-Mh) aims at being your main entrypoint for managaing all these different tools, so that you can get:
+
+1. A unified configuration framework for all of these
+2. A powerful and ergonomic language to allow you to easily abstract over these configurations, with discoverability and early error reporting
+3. A way to get these tools to communicate with each others, so that your custom scripts can easily leverage your package manager to get the dependencies they need, your CI setup can understand your build system and parallelise efficiently, and so on.
+
+## Getting started
+
+To start using `nickel-nix`, you need `Nix` to be installed and configured.
+If it isn't already the case, you can get it with:
+
+```console
+$ curl -L https://nixos.org/nix/install | bash
+# We also need a couple of experimental Nix features
+$ mkdir -p ~/.config/nix
+$ echo 'experimental-features = nix-command flakes' >> ~/.config/nix/nix.conf
+```
+
+Then bootstrap a project with:
+
+```console
+$ nix flake init -t github:nickel-lang/nickel-nix
+# Edit the project file to fit your needs
+$ $EDITOR project.ncl
+# Enter the environment
+$ . .nickel-nix/activate
+```
+
+## Managing your environment with Nixel
+
+### Dependencies
+
+Nixel can be used to declare the dependencies for your project.
+These can then be instantiated using [Nix](https://nixos.org/nix).
+
+> TODO: Document
+
+### CI configuration
+
+You can use Nixel to generate a CI configuration.
+This is extremely handy if you want to test some non-trivial matrix of platforms/configuration.
+
+Only Github actions is supported at the moment, but we plan to add more options soon.
+
+> TODO: Document
+
+### Development services
+
+Nixel can also be used for some simple service management.
+In combination with the Nix integration, this allows you to quickly spawn any service that you might depend on.
+
+Once the services have been defined in your `project.ncl` file, you can run `nixel up` to start them.
```

Comment:
Yeah, I didn't really find a way around that. But as you say, it can be made available by Nix, so it's not too bad. Besides, we need something to trigger the regeneration of the static files that we will necessary have (CI config or such)

Reply:
Unless we talk GitHub into allowing for some dynamic workflow generation :stuck_out_tongue: but yeah, realistically, we'll have to statically generate the CI files at first.

Reply:
I think we should bundle services with shells, CIs and other pieces, so that user can just declare "I write in Rust using Postgres" and let Nixel spawn Postgres locally and in "functional" test job, add "build" job that would trigger usual Rust build, and provide all necessary tools like `cargo`, `rustc` and `psql` in dev environment.